Packs/Rapid7_InsightIDR/Integrations/Rapid7_InsightIDR/Rapid7_InsightIDR.py

```diff
@@ -667,6 +631,31 @@ def insight_idr_query_log_set_command(client: Client, log_set_id: str, query: st
     return command_results
 
 
+def handle_query_log_results(client: Client, result):
+    data_for_readable_output = []
+    raw_responcse = []
+
+    results_list = [result]
+    while results_list:
+        results = results_list.pop(0)
+
+        if events := results.get('events', []):
+            data_for_readable_output.extend(events)
+            raw_responcse.append(results)
+
+        if 'links' in results:
+            for link in results.get('links', []):
+                url = link.get('href')
+                new_results = client.query_log_callback(url)
+                results_list.append(new_results)
+
+                events_len = len(results.get('events', []))
+                progress = results.get('progress')
+                demisto.debug(f'Events length: {events_len}, progress: {progress}')
```

Comment:
```suggestion
                progress = results.get('progress')
                demisto.debug(f'Events length: {len(events)}, progress: {progress}')
```
events is already assigned a value in the previous if statement